Source/Core/DolphinQt2/GameList/GameList.cpp

```diff
@@ -531,16 +549,102 @@ static bool CompressCB(const std::string& text, float percent, void* ptr)
 {
   if (ptr == nullptr)
     return false;
+
   auto* progress_dialog = static_cast<QProgressDialog*>(ptr);
 
   progress_dialog->setValue(percent * 100);
   return !progress_dialog->wasCanceled();
 }
 
+void GameList::OnSectionResized(int index, int, int)
+{
+  auto* hor_header = m_list->horizontalHeader();
+
+  std::vector<int> sections;
+
+  const int vis_index = hor_header->visualIndex(index);
+  const int col_count = hor_header->count() - hor_header->hiddenSectionCount();
+
+  bool last = true;
+
+  for (int i = vis_index + 1; i < col_count; i++)
+  {
+    if (hor_header->sectionResizeMode(i) != QHeaderView::Interactive)
+      continue;
+
+    last = false;
+    break;
+  }
+
+  if (!last)
+  {
+    for (int i = 0; i < vis_index; i++)
+    {
+      if (hor_header->sectionResizeMode(i) != QHeaderView::Interactive)
+        continue;
+
+      hor_header->setSectionResizeMode(hor_header->logicalIndex(i), QHeaderView::Fixed);
+      sections.push_back(i);
+    }
+
+    OnHeaderViewChanged();
+
+    for (int i : sections)
+    {
+      hor_header->setSectionResizeMode(hor_header->logicalIndex(i), QHeaderView::Interactive);
+    }
+  }
+  else
+  {
+    OnHeaderViewChanged();
+  }
+}
+
 void GameList::OnHeaderViewChanged()
 {
-  Settings::GetQSettings().setValue(QStringLiteral("tableheader/state"),
-                                    m_list->horizontalHeader()->saveState());
+  static bool block = false;
+
+  if (block)
+    return;
+
+  block = true;
+
+  // So here's the deal: Qt's way of resizing stuff around stretched columns sucks ass
+  // That's why instead of using Stretch, we'll just make resizable columns take all the available
+  // space ourselves!
+
+  int available_width = width() - style()->pixelMetric(QStyle::PM_ScrollBarExtent) - 5;
+  int previous_width = 0;
+
+  std::vector<int> candidate_columns;
+
+  // Iterate through all columns
+  for (int i = 0; i < GameListModel::NUM_COLS; i++)
+  {
+    if (m_list->isColumnHidden(i))
+      continue;
+
+    if (m_list->horizontalHeader()->sectionResizeMode(i) == QHeaderView::Fixed)
+    {
+      available_width -= m_list->columnWidth(i);
+    }
+    else
+    {
+      candidate_columns.push_back(i);
+      previous_width += m_list->columnWidth(i);
+    }
+  }
+
+  for (int column : candidate_columns)
```

Comment:
Add this right above the for loop: `int new_width = 0;`

Reply:
Not really sure what you're trying to tell me. There is no such variable.

Reply:
See the comment below.